src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php

```diff
@@ -52,18 +52,26 @@ public function process(ContainerBuilder $container)
             }
             $config = $resolvingBag->resolveValue($config);
 
-            $tmpContainer = new ContainerBuilder($resolvingBag);
-            $tmpContainer->setResourceTracking($container->isTrackingResources());
-            $tmpContainer->addObjectResource($extension);
-            if ($extension instanceof ConfigurationExtensionInterface && null !== $configuration = $extension->getConfiguration($config, $tmpContainer)) {
-                $tmpContainer->addObjectResource($configuration);
-            }
+            try {
+                $tmpContainer = new ContainerBuilder($resolvingBag);
+                $tmpContainer->setResourceTracking($container->isTrackingResources());
+                $tmpContainer->addObjectResource($extension);
+                if ($extension instanceof ConfigurationExtensionInterface && null !== $configuration = $extension->getConfiguration($config, $tmpContainer)) {
```

Comment:
Assignment in condition?

Reply:
Indeed :) Was here before, and not forbidden by our policy, you can find many others in the code base.